### PR TITLE
use Flow.lazyInit in Artery TCP

### PR DESCRIPTION
**Pull request by patriknw**
Merge after https://github.com/akka/akka/pull/24527

### CODE REVIEW

File: akka-remote/src/main/scala/akka/remote/artery/tcp/ArteryTcpTransport.scala [hunk 1]
```diff
@@ -124,17 +124,19 @@ private[remote] class ArteryTcpTransport(_system: ExtendedActorSystem, _provider
...
-          s"${outboundContext.remoteAddress.host.get}:${outboundContext.remoteAddress.port.get} " +
-            s"/ ${streamName(streamId)}")
-
-        // FIXME use the Flow.lazyInit from https://github.com/akka/akka/pull/24527
-
         val flow =
           Flow[ByteString]
-            .prepend(Source.single(TcpFraming.encodeConnectionHeader(streamId)))
-            .via(connectionFlow)
+            .via(Flow.lazyInit(_ ⇒ {
```
Comment (patriknw) [inline akka-remote/src/main/scala/akka/remote/artery/tcp/ArteryTcpTransport.scala]: the whole thing is wrapped in `RestartFlow.withBackoff` and then I don't want to open new connection immediately after backoff unless there is a new message

cc @gosubpl this was what I needed it for in Artery

Comment (gosubpl) [reply akka-remote/src/main/scala/akka/remote/artery/tcp/ArteryTcpTransport.scala]: cool! my use case is a bit different, similar to `switchMap`. I use it in an HTTP route, that has elements and commands. Commands result in different processing graphs (flows) for elements. So I `splitWhen` on each command, `Flow.lazyInit` build processor for subsequent elements and `concatSubstreams` finally.

**Comment by patriknw** [comment]
rebased after the merge of lazyInit

**Comment by patriknw** [comment]
failed to bind to localhost/127.0.0.1:37893
address in use

**Comment by patriknw** [comment]
PLS BUILD

**Review by johanandren — APPROVED** [review]
LGTM

**Review by ktoso — APPROVED** [review]

**Comment by ktoso** [comment]
Nice!

**Comment by gosubpl** [comment]
🎉

**Pull request merged**